src/docker/composebuilder/service_builder_test.go

```diff
@@ -0,0 +1,225 @@
+		var dockerConfigs types.DockerConfigs
+		var serviceEndpoints map[string]*composebuilder.ServiceEndpoints
+
+		var _ = BeforeEach(func() {
+			appDir := helpers.GetTestApplicationDir("service-specific-dependency")
+			appConfig, err := types.NewAppConfig(appDir)
+			Expect(err).NotTo(HaveOccurred())
+			serviceConfigs, err := config.GetServiceConfigs(appDir, appConfig)
+			Expect(err).NotTo(HaveOccurred())
+			serviceData := appConfig.Services
+			serviceRole := "postgres-service"
+			buildMode := composebuilder.BuildMode{
+				Type:        composebuilder.BuildModeTypeLocal,
+				Environment: composebuilder.BuildModeEnvironmentDevelopment,
+			}
+			serviceEndpoints = map[string]*composebuilder.ServiceEndpoints{
+				"postgres-service": &composebuilder.ServiceEndpoints{},
+			}
+			dockerConfigs, err = composebuilder.GetServiceDockerConfig(appConfig, serviceConfigs[serviceRole], serviceData[serviceRole], serviceRole, appDir, buildMode, serviceEndpoints)
+			Expect(err).NotTo(HaveOccurred())
+		})
+
+		It("should pass dependency env variables to services", func() {
+			postgresServiceDockerConfig, exists := dockerConfigs["postgres-service"]
+			Expect(exists).To(Equal(true))
+			Expect(postgresServiceDockerConfig.Environment["DB_NAME"]).To(Equal("my_db"))
+		})
+	})
+})
+
+var _ = Describe("building for local production", func() {
+	var dockerConfigs types.DockerConfigs
+	var appDir string
+	var serviceEndpoints map[string]*composebuilder.ServiceEndpoints
+
+	var _ = BeforeEach(func() {
+		appDir = helpers.GetTestApplicationDir("simple")
+		appConfig, err := types.NewAppConfig(appDir)
+		Expect(err).NotTo(HaveOccurred())
+		serviceConfigs, err := config.GetServiceConfigs(appDir, appConfig)
+		Expect(err).NotTo(HaveOccurred())
+		serviceData := appConfig.Services
+		serviceRole := "web"
+		buildMode := composebuilder.BuildMode{
+			Type:        composebuilder.BuildModeTypeLocal,
+			Mount:       true,
+			Environment: composebuilder.BuildModeEnvironmentProduction,
+		}
+		serviceEndpoints = map[string]*composebuilder.ServiceEndpoints{
+			"web": &composebuilder.ServiceEndpoints{},
+		}
+		dockerConfigs, err = composebuilder.GetServiceDockerConfig(appConfig, serviceConfigs[serviceRole], serviceData[serviceRole], serviceRole, appDir, buildMode, serviceEndpoints)
+		Expect(err).NotTo(HaveOccurred())
+	})
+
+	It("sets the correct dockerfile", func() {
+		dockerConfig, exists := dockerConfigs["web"]
+		Expect(exists).To(Equal(true))
+		Expect(dockerConfig).To(Equal(types.DockerConfig{
+			Build: map[string]string{
+				"dockerfile": "Dockerfile.prod",
+				"context":    "${APP_PATH}/web",
+			},
+			ContainerName: "web",
+			Command:       "",
+			Ports:         []string{},
+			Volumes:       []string{"${APP_PATH}/web:/mnt"},
+			Environment: map[string]string{
+				"ROLE":        "web",
+				"EXOCOM_HOST": "exocom0.26.1",
+			},
+			DependsOn: []string{"exocom0.26.1"},
+			Restart:   "on-failure",
+		}))
+	})
+})
```

Comment:
Moved from `service_test.go` -> `service_builder_test.go`, and deleted some outdated tests (see below)